proposal/custom-url-resolution.md

```diff
@@ -0,0 +1,108 @@
+
+## Summary
+
+> This section is non-normative.
+
+This proposal defines a standardized way to remap the url references to the final location on the server or inline reference.
+
+This is accomplished by running url references through the url rewriting plugin if a url rewriting plugin has been defined.
+
+### Steps
+
+The steps of rewriting a url reference:
+
+- Url reference gets extracted from the sass file (For more information see [Semantics](#semantics))
+- These reference gets passed into the rewrite url plugin which returns the output url of this resource.
+- This returned value gets used to replace the original url reference.
+
+### JavaScript API
+
+At the core of remapping the url references is the JavaScript API which allows users to return a new url reference, based on filepath and original url reference.
+
+The first parameter of the urlRewrite function is an object with the following values:
+
+- `file`: The file path of the sass file that references the url.
+- `entry`: The entry file path that has been used to call the sass compiler.
+- `url`: The original url reference, for example with `url(file://../assets/test.png)` it would return be `file://../assets/test.png`.
+
+The second parameter of the urlRewrite function is an optional done callback that is used when performing asynchronous operations.
+
+This callback function takes in two parameters:
+
+- `error`: This can be null or an Error object, `null` means there is no error.
+- `url`: This is the `string` that gets used to replace the url reference.
+
+Asynchronous example:
+
+```TypeScript
+let sassOptions = {
+  // Rewrite urls asynchronously
+  rewriteUrl: async (importerResult: { file: string, entry: string, url: string }, done: (error: Error | null, url: string | null) => void): void => {
+    if (file && !content) {
+      content = await fs.readFile(path.join(path.dirname(file), url));
+    }
+
+    done(null, content ? `data:${base64(content)))}` : url);
+  }
+}
+```
+
+Synchronous example:
+
+```TypeScript
+let sassOptions = {
+  // Rewrite urls synchronously
+  rewriteUrl: (importerResult: { file: string, entry: string, url: string }): string | null => {
+    if (!importerResult.file) {
+      return url;
+    }
+
+    return path.relative(path.dirname(entry), path.join(path.dirname(file), url));
+  }
+}
+```
+
+## Semantics
+
+This proposal defines an overload for `url()`, to allow users to rewrite url references to how it will be accessible on the browser.
+
+It applies to both with and without quotation marks: `url("$url")` and `url($url)`.
+
+### Using Variables
+
+Variables can be used in a url reference, in this case it should get remapped based on the value that gets created after the variables have been applied to ensure we are able to remap the url as expected.
+
+Examples:
+
+```Scss
+url("#{$asset-path}/image.png");
+```
+
+```Scss
+url("./folder/#{$some-var}");
+```
+
+## Deprecation Process
```

Comment:
You don't need to have a "deprecation process" section if you're not introducing any breaking changes to the Sass API.

Reply:
You're probably right, this is breaking but opt-in so will remove it